lib/config/target.js

```diff
@@ -0,0 +1,299 @@
+ * @property {boolean | null} node node platform, require of node built-in modules is available
+ * @property {boolean | null} nwjs nwjs platform, require of legacy nw.gui is available
+ * @property {boolean | null} electron electron platform, require of some electron built-in modules is available
+ */
+
+/**
+ * @typedef {Object} ElectronContextTargetProperties
+ * @property {boolean | null} electronMain in main context
+ * @property {boolean | null} electronPreload in preload context
+ * @property {boolean | null} electronRenderer in renderer context with node integration
+ */
+
+/**
+ * @typedef {Object} ApiTargetProperties
+ * @property {boolean | null} require has require function available
+ * @property {boolean | null} nodeBuiltins has node.js built-in modules available
+ * @property {boolean | null} document has document available (allows script tags)
+ * @property {boolean | null} importScripts has importScripts available
+ * @property {boolean | null} importScriptsInWorker has importScripts available when creating a worker
+ * @property {boolean | null} fetchWasm has fetch function available for WebAssembly
+ * @property {boolean | null} global has global variable available
+ */
+
+/**
+ * @typedef {Object} EcmaTargetProperties
+ * @property {boolean | null} globalThis has globalThis variable available
+ * @property {boolean | null} bigIntLiteral big int literal syntax is available
+ * @property {boolean | null} const const and let variable declarations are available
+ * @property {boolean | null} arrowFunction arrow functions are available
+ * @property {boolean | null} forOf for of iteration is available
+ * @property {boolean | null} destructuring destructuring is available
+ * @property {boolean | null} import async import() is available
+ * @property {boolean | null} importInWorker async import() is available when creating a worker
+ * @property {boolean | null} module ESM syntax is available (when in module)
+ */
+
+///** @typedef {PlatformTargetProperties | ApiTargetProperties | EcmaTargetProperties | PlatformTargetProperties & ApiTargetProperties | PlatformTargetProperties & EcmaTargetProperties | ApiTargetProperties & EcmaTargetProperties} TargetProperties */
+/** @template T @typedef {{ [P in keyof T]?: never }} Never<T> */
+/** @template A @template B @typedef {(A & Never<B>) | (Never<A> & B) | (A & B)} Mix<A,B> */
+/** @typedef {Mix<Mix<PlatformTargetProperties, ElectronContextTargetProperties>, Mix<ApiTargetProperties, EcmaTargetProperties>>} TargetProperties */
+
+const versionDependent = (major, minor) => {
+	if (!major) return () => /** @type {undefined} */ (undefined);
+	major = +major;
+	minor = minor ? +minor : 0;
+	return (vMajor, vMinor = 0) => {
+		return major > vMajor || (major === vMajor && minor >= vMinor);
+	};
+};
+
+/** @type {[string, string, RegExp, (...args: string[]) => TargetProperties][]} */
+const TARGETS = [
+	[
+		"web",
+		"Web browser.",
```

Comment:
I think better avoid this long string, It seems to me that no one will use them, `web` is short and simple